cml-tdengine/src/core/inference.rs

```diff
@@ -48,15 +46,17 @@ impl<D: IntoDsn + Clone> Inference<Field, Value, Manager<TaosBuilder>> for TDeng
         inference_fn: FN,
     ) -> Result<()>
     where
-        FN: FnOnce(&mut Vec<NewSample<Value>>, &Pool<TaosBuilder>) -> R,
-        R: Future<Output = Result<Vec<NewSample<Value>>>>,
+        FN: FnOnce(&mut Vec<NewSample<Value>>, &str, i64) -> Vec<NewSample<Value>>,
     {
-        let samples_with_res = inference_fn(data, pool).await?;
-
         let taos = pool.get().await?;
-        taos.use_database("training_data").await?;
         let mut stmt = Stmt::init(&taos)?;
-
+        taos.use_database("task").await?;
+        let last_task_time = taos
+            .query_one(format!("SELECT LAST(ts) FROM task.`{}`", metadata.batch()))
```

Comment:
Here we need to pick the task according to `available_status`.

BTW, `query_one` is a new API which is a better option for these scenarios. Could you replace the original code with it in other appropriate places?